src/webgpu/util/math.ts

```diff
@@ -288,16 +295,57 @@ export function linearRange(a: number, b: number, num_steps: number): Array<numb
  * thus biasing towards 0, but remaining on the [0, 1] range.
  * This biased range is then scaled to the desired range using lerp.
  * Different curves could be generated by changing c, where greater values of c will bias more towards 0.
- * */
+ */
 export function biasedRange(a: number, b: number, num_steps: number): Array<number> {
   const c = 2;
-  assert(num_steps > 0, '|num_steps| must be greater than 0');
+  if (num_steps <= 0) {
+    return Array<number>();
+  }
+
+  // Avoid division by 0
+  if (num_steps === 1) {
+    return [a];
+  }
 
   return Array.from(Array(num_steps).keys()).map(i =>
     lerp(a, b, Math.pow(lerp(0, 1, i / (num_steps - 1)), c))
   );
 }
 
+/**
+ * @returns an ascending sorted array of numbers spread over the entire range of 32-bit floats
+ *
+ * Numbers are divided into 4 regions: negative normals, negative subnormals, positive subnormals & positive normals.
+ * Zero is included. The normal number regions are biased towards zero, and the subnormal regions are linearly spread.
+ *
+ * @param neg_count number of numbers to generate in the negative normals region
+ * @param neg_sub_count number of numbers to generate in the negative subnormals region
+ * @param pos_sub_count number of numbers to generate in the positive subnormals region
+ * @param pos_count number of numbers to generate in the positive normals region
+ */
+export function fullF32Range(
+  neg_count: number = 50,
```

Comment:
Observation: Currently you always use `pos_sub_count` == `neg_sub_count`, and `pos_count` == `neg_count`.

If you were to go with the structure approach, you could have the 'neg' fields default to the positive ones, reducing usage boilerplate:

```ts
export function fullF32Range(
  rng: {
    neg_count?: number; // defaults to count
    neg_sub_count?: number; // defaults to sub_count
    sub_count: number;
    count: number;
  } = {
    sub_count: 10,
    count: 50,
  }
): Array<number> {
  const neg_count = rng.neg_count === undefined ? rng.count : (rng.neg_count as number);
  const neg_sub_count =
    rng.neg_sub_count === undefined ? rng.sub_count : (rng.neg_sub_count as number);

  return [
    ...biasedRange(kValue.f32.negative.max, kValue.f32.negative.min, neg_count),
    ...linearRange(
      kValue.f32.subnormal.negative.min,
      kValue.f32.subnormal.negative.max,
      neg_sub_count
    ),
    0.0,
    ...linearRange(
      kValue.f32.subnormal.positive.min,
      kValue.f32.subnormal.positive.max,
      rng.sub_count
    ),
    ...biasedRange(kValue.f32.positive.min, kValue.f32.positive.max, rng.count),
  ];
}
```